arch/xtensa/src/esp32/Kconfig

```diff
@@ -1158,6 +1158,37 @@ config ESP32_SPI2_MISOPIN
 	default 12
 	range 0 39
 
+choice ESP32_SPI2_MASTER_IO
+	prompt "SPI2 master I/O mode"
+	default ESP32_SPI2_MASTER_IO_RW
```

Comment:
Should it have a "depends on !SPI_SLAVE" ? Otherwise the user could select the ESP32_SPI2_MASTER_IO even when SPI_SLAVE is enabled

Reply:
It may not depend on "SPI_SLAVE", because users may use SPI2 as SPI master and SPI3 as SPI slave, so we should keep SPI2 or SPI3 master mode when selecting SPI_SLAVE.

Reply:
but the current design allow use select ESP32_SPI2_MASTER_IO_RW and ESP32_SPI2_SLAVE_IO_RW, which looks wrong. You should merge ESP32_SPI2_MASTER_IO_RW and ESP32_SPI2_SLAVE_IO_RW into one choice to avoid the invalid combination.

Reply:
It should have been designed as you said, but because ESP32 has not enough GPIOs and peripherals, so some hardware needs to switch connection from SPI master to slave, like:
1. ESP32 SPI0 & 1 connect to SPI flash and PSRAM
2. ESP32 SPI2 connects to some sensors
1. ESP32 SPI3 works as master mode and configure the camera
3. ESP32 SPI3 switches to slave mode, then receive data from camera
So in the case, we have to make the master and slave mode co-exist in specific GPIOs and peripheral.

Reply:
The most correct would be to move these kind of configuration out of Kconfig, and place them into the board headers. But this is not a simple change as it would affect several other peripherals that already do pin mapping via Kconfig.

Considering the use case scenario described by @donghengqaz, the current implementation seems fit.